### ISSUE TITLE
Incorrect patch when replacing object value.

### ISSUE THREAD

**Issue by jdalton** [Rich-Harris/golden-fleece#5]
```js
console.log(fleece.patch('{scripts:{a:"",b:""}}', {scripts:{c:"c"}}))
// => {scripts:, c: "c"}}
```

**Comment by Conduitry** [comment]
Opened PR #9

**Comment by jdalton** [comment]
Thank you @Conduitry!

**Comment by Rich-Harris** [comment]
You know it's a @Conduitry PR when it fixes the bug *and* reduces the LOC count 😀 Released 1.0.6 with the fix — thanks

**Comment by jdalton** [comment]
@Rich-Harris I saw the reduced LOC and was like whoa too! 👏
